experimental/torch_xla2/test/test_unbounded_dynamism.py

```diff
@@ -0,0 +1,662 @@
+import re
+import sys
+import unittest
+
+import numpy as np
+import torch
+from torch.export import Dim, export
+from torch_xla2.export import exported_program_to_stablehlo as exp2shlo
+
+## This file is copied from `xla/test/stablehlo/test_unbounded_dynamism.py`
+## To test that torch_xla2 has identical behavior.
+## The only differences in this test files are that torch_xla2 export preserves
+## argument order more often than torch_xla export.
+##
+## This broke ~5 tests, for example: test_bmm_dynamic_out_dim
+##   args = (
+##     torch.rand((8, 128, 256)),
+##     torch.rand((8, 256, 3)),
+##   )
+##   dynamic_shapes = ((None, {2: Dim("dim")}),)
+##   ...
+##   torch_xla_regex = r'%arg.: tensor<8x256x\?xf32>.*%arg.: tensor<8x128x256xf32>.*->.*tensor<8x128x\?xf32>'
+##   torch_xla2_regex = r'%arg.: tensor<8x128x256xf32>.*%arg.: tensor<8x256x\?xf32>.*->.*tensor<8x128x\?xf32>'
+
+# Shim to run tests
+class ExportAdapter():
+  def __init__(self, export):
+    self.export = export
+
+  def get_stablehlo_text(self):
+    return self.export.mlir_module()
+
+def exported_program_to_stablehlo(exported):
+  return ExportAdapter(exp2shlo(exported))
+
+def wrap_func_as_nn_module(f):
+  class M(torch.nn.Module):
+    def __init__(self):
+      super().__init__()
+      
+    def forward(self, *args):
+      return f(*args)
+  return M().eval()
+
+class UnboundedDynamismExportTest(unittest.TestCase):
+
+  def test_add(self):
```

Comment:
Most of these tests look very similar. Consider extracting the duplicated bits to a function.

Reply:
this was the forked test, going to keep it as-is to be aligned with original (easy diffing).